okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -253,9 +255,10 @@ public OkHttpClientStream newStream(MethodDescriptor<?, ?> method,
         Headers.createRequestHeaders(headers, defaultPath, defaultAuthority);
 
     SettableFuture<Void> pendingFuture = null;
+    boolean shouldGoAway = false;
     synchronized (lock) {
```

Comment:
startStream() calls into the stream _and_ modifies state protected by lock (including `frameWriter.synStream()`).